xrpl/asyncio/transaction/main.py

```
@@ -498,3 +520,66 @@ async def _fetch_owner_reserve_fee(client: Client) -> int:
+                account_sequences[raw_txn.account] += 1
+            else:
+                sequence = await get_next_valid_seq_number(raw_txn.account, client)
+                account_sequences[raw_txn.account] = sequence + 1
+                raw_txn_dict["sequence"] = sequence
+
+        if raw_txn.is_signed():
+            raise XRPLException("Inner Batch transactions must not be signed.")
+
+        # validate fields that are supposed to be empty/zeroed
+        def _validate_field(field_name: str, expected_value: Union[str, None]) -> None:
+            if field_name not in raw_txn_dict:
```

Ruff (0.11.9)

Function definition does not bind loop variable `raw_txn_dict`

(B023)